packages/protocol/contracts/governance/DowntimeSlasher.sol

```diff
@@ -35,88 +42,212 @@ contract DowntimeSlasher is SlasherUtil {
+   * @notice Calculates and returns the signature bitmap for the specified interval.
+   * Similar to the parentSealBitmap of every block (where you have which validators were
+   * able to sign the previous block), this bitmap shows for that specific interval which
+   * validators signed at least one block
+   * @param startBlock First block of the downtime interval.
+   * @param endBlock Last block of the downtime interval.
+   * @return The signature uptime bitmap for the specified interval.
+   * @dev startBlock and endBlock must be in the same epoch.
+   * @dev The getParentSealBitmap precompile requires that startBlock must be within 4 epochs of 
+   * the current block.
+   */
+  function getBitmapForInterval(uint256 startBlock, uint256 endBlock)
+    public
+    view
+    returns (bytes32)
+  {
+    require(endBlock >= startBlock, "endBlock must be greater or equal than startBlock");
+    // The sealBitmap for the block N is stored in the block N+1
+    // (block.number - 1) is the currentBlock, and because we don't have the sealBitmap for
+    // the currentBlock, we should check against (currentBlock - 1).
+    uint256 lastBlockWithParentSealBitmap = block.number.sub(2);
+    require(
+      endBlock <= lastBlockWithParentSealBitmap,
+      "endBlock's parentSealBitmap is not yet available"
+    );
+    uint256 epochSize = getEpochSize();
+    require(
+      lastBlockWithParentSealBitmap.sub(startBlock) <= epochSize.mul(4),
+      "startBlock must be within 4 epochs of the current head"
+    );
+    require(
+      epochNumberOfBlock(startBlock, epochSize) == epochNumberOfBlock(endBlock, epochSize),
+      "startBlock and endBlock must be in the same epoch"
+    );
+
+    bytes32 bitmap;
+    for (uint256 n = startBlock; n <= endBlock; n++) {
+      // The canonical signatures for block N are stored in the parent seal bitmap for block N+1.
+      bitmap |= getParentSealBitmap(n.add(1));
+    }
+
+    return bitmap;
+  }
+
+  /**
+   * @notice Calculates and sets the signature bitmap for the specified interval.
+   * @param startBlock First block of the downtime interval.
+   * @param endBlock Last block of the downtime interval.
+   * @return The signature bitmap for the specified interval.
+   * @dev startBlock and endBlock must be in the same epoch.
+   */
+  function setBitmapForInterval(uint256 startBlock, uint256 endBlock) public returns (bytes32) {
+    require(!isBitmapSetForInterval(startBlock, endBlock), "bitmap already set");
+
+    bytes32 bitmap = getBitmapForInterval(startBlock, endBlock);
+    bitmaps[msg.sender][startBlock][endBlock] = bitmap;
+
+    emit BitmapSetForInterval(startBlock, endBlock, bitmap);
+
+    return bitmap;
+  }
+
+  /**
+   * @notice Check if a validator appears down in the bitmap for the interval of blocks.
+   * Both startBlock and endBlock should be part of the same epoch.
+   * @param startBlock First block of the interval.
```

Comment:
nit: `startBlock` and `endBlock` are parameters to a lot of these functions, but have different descriptions across the natspecs. Suggest using these for every function for consistency:
```
   * @param startBlock First block of the interval.
   * @param endBlock Last block of the interval.
```